py_mmd_tools/nc_to_mmd.py

```diff
@@ -0,0 +1,328 @@
+                dacc_desc = ET.SubElement(dacc, ET.QName(ns_map['mmd'], 'description'))
+                dacc_desc.text = str(desc)
+                dacc_res = ET.SubElement(dacc, ET.QName(ns_map['mmd'], 'resource'))
+                if 'OGC WMS' in prot_type:
+                    wms_layers = ET.SubElement(dacc, ET.QName(ns_map['mmd'], 'wms_layers'))
+                    # Don't add variables containing these names to the wms layers
+                    skip_layers = ['latitude', 'longitude', 'angle']
+                    for w_layer in all_netcdf_variables:
+                        if any(skip_layer in w_layer for skip_layer in skip_layers):
+                            continue
+                        wms_layer = ET.SubElement(wms_layers, ET.QName(ns_map['mmd'], 'wms_layer'))
+                        wms_layer.text = w_layer
+                    # Need to add get capabilities to the wms resource
+                    res += '?service=WMS&version=1.3.0&request=GetCapabilities'
+                dacc_res.text = res
+
+        # Add OGC WMS data_access as comment
+        root.append(ET.Comment(str('<mmd:data_access>\n\t<mmd:type>OGC WMS</mmd:type>\n\t<mmd:description>OGC Web '
+                                   'Mapping Service, URI to GetCapabilities Document.</mmd:description>\n\t'
+                                   '<mmd:resource></mmd:resource>\n\t<mmd:wms_layers>\n\t\t<mmd:wms_layer>'
+                                   '</mmd:wms_layer>\n\t</mmd:wms_layers>\n</mmd:data_access>')))
+
+        et = ET.ElementTree(ET.fromstring(ET.tostring(root, pretty_print=True).decode("utf-8")))
+        et.write(str(self.output_file), pretty_print=True)
+
+    def required_mmd_elements(self):
+        """ Create dict with required MMD elements"""
+
+        mmd_required_elements = {
+                'metadata_identifier': None,
+                'metadata_status': None,
+                'collection': None,
+                'title': None,
+                'abstract': None,
+                'last_metadata_update': None,
+                'dataset_production_status': None,
+                'operational_status': None,
+                'iso_topic_category': None,
+                'keywords': '\n\t\t<mmd:keyword></mmd:keyword>\n\t',
+                'temporal_extent': '\n\t\t<mmd:start_date></mmd:start_date>\n\t',
+                'geographic_extent': str('\n\t\t<mmd:rectangle srsName=""> \n\t\t\t<mmd:north></mmd:north> \n\t\t\t'
+                                         '<mmd:south></mmd:south> \n\t\t\t<mmd:east></mmd:east> \n\t\t\t<mmd:west>'
+                                         '</mmd:west> \n\t\t</mmd:rectangle>\n\t')
+            }
+        return mmd_required_elements
+
+    def generate_cf_mmd_lut_missing_acdd(self):
+        """ Translation dict between ACDD and MMD (dict={ACDD: MDD})
+
+        Create lookup table for mandatory MMD elements missing in the ACDD
+        but that still is present as global attributes in the netCDF file"""
```

Comment:
What is meant by "but that still is present as global attributes in the netCDF file"? What standard are the dictionary values following? It it GCMD? Has the original code for this been made for some specific netcdf files?

Reply:
As mentioned on phone I think some of these are remains of the original Damocles setup we were using. Not everything is captured in ACDD, but the software should combine all elements into a single configuration table regardless of the origin (ACDD or Damocles). This table should somehow map between global attributes and their MMD implementation. The challenge is to do this and keep track of where only one element is allowed. Preferably this should be implemented as a configuration file, but at least using some sort of YAML approach  would probably be beneficial.